src/java/com/zimbra/oauth/handlers/impl/GoogleContactsImport.java

```diff
@@ -0,0 +1,833 @@
+    }
+
+    /**
+     * Requests contacts for the given credentials.
+     *
+     * @param url The contacts url
+     * @param authorizationHeader The credentials header
+     * @return Json contacts response
+     * @throws ServiceException If there are issues retrieving the data
+     * @throws IOException If there are issues executing the request
+     */
+    protected JsonNode getContactsRequest(String url, String authorizationHeader)
+        throws ServiceException, IOException {
+        final GetMethod get = new GetMethod(url);
+        get.addRequestHeader(OAuth2Constants.HEADER_AUTHORIZATION, authorizationHeader);
+        ZimbraLog.extensions.debug("Fetching contacts for import.");
+        return OAuth2Handler.executeRequestForJson(get);
+    }
+
+    /**
+     * Retrieves a set of the contacts identifiers that exist in a specified
+     * folder.
+     *
+     * @param mailbox The mailbox
+     * @param folderId The folder
+     * @return Set of resourceNames for existing contacts
+     * @throws ServiceException If there are issues fetching the contacts
+     */
+    protected Set<String> getExistingContacts(Mailbox mailbox, int folderId)
+        throws ServiceException {
+        // fetch the list of existing contacts for the specified folder
+        List<Contact> contacts = null;
+        try {
+            contacts = mailbox.getContactList(null, folderId);
+        } catch (final ServiceException e) {
+            ZimbraLog.extensions.errorQuietly(
+                "Failed to retrieve existing contacts during Google contact sync.", e);
+            throw ServiceException
+                .FAILURE("Failed to retrieve existing contacts during Google contact sync.", e);
+        }
+
+        // create a resourceName set
+        final Set<String> contactsIdentifiers = new HashSet<String>();
+        for (final Contact contact : contacts) {
+            if (contact != null) {
+                final String resourceName = contact.get("resourceName");
+                if (resourceName != null) {
+                    contactsIdentifiers.add(resourceName);
+                }
+            }
+        }
+
+        return contactsIdentifiers;
+    }
+
+    /**
+     * Processes json contacts from google api into a ParsedContact, adding it
+     * to a create list if the contact does not already exist in the datasource
+     * folder - based on the resourceName property.
+     *
+     * @param existingContacts Contact resourceNames in the datasource folder
+     * @param jsonContacts Json contacts from google api
+     * @param createList List of contacts to create
+     */
+    protected void parseNewContacts(Set<String> existingContacts, JsonNode jsonContacts,
+        List<ParsedContact> createList) {
+        for (final JsonNode contactElement : jsonContacts) {
+            try {
+                ZimbraLog.extensions
+                    .debug("Verifying if new contact for: " + jsonContacts.toString());
```

Comment:
Please use ZimbraLog.extensions
+                    .debug("Verifying if new contact for: %s" , jsonContacts.toString());